app/models/announcement.rb

```diff
@@ -1,2 +1,8 @@
 class Announcement < ApplicationRecord
+  VALID_BANNER_STYLES = %w[default brand success warning error].freeze
+  resourcify
+
+  has_one :broadcast, as: :broadcastable
```

Comment:
I think we probably need to validate that an `Announcement` always has a `Broadcast`; right now, you can manually do this and it won't fail:
```ruby
[20] pry(main)> Announcement.new(banner_style: 'brand').save
   (0.6ms)  BEGIN
  Announcement Create (18.7ms)  INSERT INTO "announcements" ("banner_style", "created_at", "updated_at") VALUES ($1, $2, $3) RETURNING "id"  [["banner_style", "brand"], ["created_at", "2020-07-28 00:03:58.241236"], ["updated_at", "2020-07-28 00:03:58.241236"]]
   (5.9ms)  COMMIT
=> true
```

...but that will result in an orphaned `Announcement`:
```ruby
[21] pry(main)> Announcement.last.broadcast
  Announcement Load (0.7ms)  SELECT "announcements".* FROM "announcements" ORDER BY "announcements"."id" DESC LIMIT $1  [["LIMIT", 1]]
  Broadcast Load (0.4ms)  SELECT "broadcasts".* FROM "broadcasts" WHERE "broadcasts"."broadcastable_id" = $1 AND "broadcasts"."broadcastable_type" = $2 LIMIT $3  [["broadcastable_id", 2], ["broadcastable_type", "Announcement"], ["LIMIT", 1]]
=> nil
```